apps/rehabstod/src/components/Table/TableLayout.tsx

```diff
@@ -0,0 +1,69 @@
+import { useNavigate } from 'react-router-dom'
+import { ReactNode, useEffect } from 'react'
+import { IDSButton } from '@frontend/ids-react-ts'
+import { FetchBaseQueryError } from '@reduxjs/toolkit/query'
+import { SerializedError } from '@reduxjs/toolkit'
+import { User } from '../../schemas'
+import { UnansweredCommunicationError } from '../error/UnansweredCommunicationError/UnansweredCommunicationError'
+import { GetTableContentError } from '../error/GetTableContentError/GetTableContentError'
+
+export function TableLayout({
+  isUserLoading,
+  user,
+  heading,
+  filters,
+  tableInfo,
+  modifyTableColumns,
+  error,
+  unansweredCommunicationError,
+  printable,
+  tableName,
+  children,
+}: {
+  isUserLoading: boolean
+  user: User | undefined
+  heading: ReactNode
+  filters: ReactNode
+  tableInfo: ReactNode
+  modifyTableColumns: ReactNode
+  error?: (FetchBaseQueryError & { id?: string }) | (SerializedError & { id?: string })
+  unansweredCommunicationError?: boolean
+  tableName: string
+  printable: boolean
+  children: ReactNode
+}) {
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    if (!isUserLoading && !user) {
+      navigate('/')
+    }
+  }, [user, isUserLoading, navigate])
```

Comment:
This could be moved to a shared hook instead feels out of place for a general purpose component.